base_layer/core/tests/tests/header_sync.rs

```diff
@@ -0,0 +1,373 @@
+                panic!("Should be 'InSyncOrAhead'");
+            }
+        },
+        _ => panic!("Expected HeadersSynchronized event"),
+    }
+    // All is good, Bob is not banned
+    assert!(!sync::wait_for_is_peer_banned(&alice_node, bob_node.node_identity.node_id(), 1).await);
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_header_sync_uneven_headers_and_blocks_peer_lies_about_pow_no_ban() {
+    // env_logger::init(); // Set `$env:RUST_LOG = "trace"`
+
+    // Create the network with Alice node and Bob node
+    let (mut alice_state_machine, alice_node, bob_node, initial_block, consensus_manager, key_manager) =
+        sync::create_network_with_alice_and_bob_nodes().await;
+
+    // Add blocks and headers to Bob's chain, with more headers than blocks
+    let blocks = sync::create_and_add_some_blocks(
+        &bob_node,
+        &initial_block,
+        10,
+        &consensus_manager,
+        &key_manager,
+        &[3; 10],
+    )
+    .await;
+    sync::delete_some_blocks_and_headers(&blocks[5..=10], WhatToDelete::Blocks, &bob_node, Some(true));
+    sync::delete_some_blocks_and_headers(&blocks[7..=10], WhatToDelete::Headers, &bob_node, None);
+    assert_eq!(bob_node.blockchain_db.get_height().unwrap(), 5);
+    assert_eq!(bob_node.blockchain_db.fetch_last_header().unwrap().height, 7);
+
+    // Add blocks and headers to Alice's chain, with more headers than blocks
+    sync::add_some_existing_blocks(&blocks[1..=10], &alice_node);
+    sync::delete_some_blocks_and_headers(&blocks[2..=10], WhatToDelete::Blocks, &alice_node, Some(true));
+    assert_eq!(alice_node.blockchain_db.get_height().unwrap(), 2);
+    assert_eq!(alice_node.blockchain_db.fetch_last_header().unwrap().height, 10);
+
+    // Alice attempts header sync, her headers are ahead, but Bob will lie about his POW
+    // Note: This behaviour is undetected!
+    let mut header_sync = sync::sync_headers_initialize_with_ping_pong_data(&alice_node, &bob_node);
+    // Remove blocks from Bpb's chain so his claimed metadata is better than what it actually is
+    sync::delete_some_blocks_and_headers(&blocks[4..=5], WhatToDelete::Blocks, &bob_node, Some(true));
+    assert!(
+        header_sync.clone().into_sync_peers()[0]
+            .claimed_chain_metadata()
+            .accumulated_difficulty() >
+            bob_node
+                .blockchain_db
+                .get_chain_metadata()
+                .unwrap()
+                .accumulated_difficulty()
+    );
+    let event = sync::sync_headers_execute(&mut alice_state_machine, &mut header_sync).await;
+    match event {
+        StateEvent::HeadersSynchronized(_val, sync_result) => {
+            assert_eq!(sync_result.headers_returned, 0);
+            assert_eq!(sync_result.peer_fork_hash_index, 3);
+            if let HeaderSyncStatus::InSyncOrAhead = sync_result.header_sync_status {
+                // Note: This behaviour is undetected! Bob cannot be banned.
```

Comment:
```suggestion
                // Note: This behaviour is undetected! Bob cannot be banned here, and should be banned by block sync
```
This test and comment makes it seem like this is a bug and a viable attack. We should probably add a test for block sync where we test this.

Reply:
Agreed, it is a viable attack, and cannot be detected by header sync alone